symengine/pow.cpp

```diff
@@ -94,7 +94,12 @@ RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
         return pownum(rcp_static_cast<const Number>(b), zero);
     }
     if (eq(*b, *one)) return a;
-    if (eq(*a, *zero)) return zero;
+    if (eq(*a, *zero) && is_a_Number(*b)) {
```

Comment:
I think this should automatically evaluate to 0.
We do such operations now as well. For example `x/x` is simplified to 1. `0*x` is simplified to `0`.